app/models/assignment.rb

```diff
@@ -24,11 +24,30 @@ def to_hash
       Hash[all.includes(:split).map { |a| [a.split.name.to_sym, a.variant.to_sym] }]
     end
 
-    def for_presentation(built_at: nil)
-      q = joins(:split)
+    def for_presentation(app_build: nil)
```

Comment:
Rather than pass a bag of disjunct args at this method, I realized a type representing an app build was better - you either aren't presenting for a specific app build, or you are, and you care about version, build date, and which app is in play.